apmpackage/cmd/genpackage/transform.go

```diff
@@ -101,6 +104,26 @@ func transformIngestPipeline(path string, content []byte, version *version.V) ([
 	return append([]byte("---\n"), content...), nil
 }
 
+func transformChangelog(content []byte, version *version.V) ([]byte, error) {
+	var doc yaml.Node
+	if err := yaml.Unmarshal(content, &doc); err != nil {
+		return nil, err
+	}
+
+	n := doc.Content[0]
+	if n.Kind != yaml.SequenceNode {
+		panic(fmt.Sprintf("expected node kind %v, got %v", yaml.SequenceNode, n.Kind))
```

Comment:
Nit. Make this a normal error? This could be caused by a typo in the changelog.
```suggestion
		return nil, fmt.Errorf("expected list of versions (%v), got %v", yaml.SequenceNode, n.Kind)
```

Reply:
I guess it's consistent with the rest of the source code. Let's leave it for the APM to discuss.

Reply:
Let's return an error. I don't think it is a big deal, this is only used for code generation. The other existing case where we panic is in code which doesn't return an error, and returning an error would make the code a lot more tedious for negligible gain.

Reply:
Fixed